ultravox/data/datasets.py

```diff
@@ -681,6 +684,80 @@ def _get_sample(self, idx, row) -> VoiceSample:
         return self._get_transcribe_sample(idx, row, tcol="text")
 
 
+class HiSantaDataset(data.IterableDataset):
+    """
+    A proprietary dataset from post-processed conversations with Santa and
+    friends between 12/18 and 12/31 2023.
+    """
+
+    class Subset(str, enum.Enum):
+        ALL = "all"  # All recoverable data
+        BEST = "best"  # Only the samples expected to have the best audio alignment
+
+    def __init__(self, args: VoiceDatasetArgs, subset: Subset = Subset.BEST) -> None:
+        super().__init__()
+        self._args = args  # TODO(mdepinet): Respect whatever args we need to.
+        self._bucket = gcs.Client().get_bucket("hisanta-dataset")
+
+        self._conversations: list[str] = json.loads(
+            self._bucket.get_blob(f"{subset}.json").download_as_bytes()
+        )["conversations"]
+        """List of references to conversation metadata JSON files in the bucket.
+        These all look like {conversation_id}/metadata.json."""
+
+        self._system_prompts: list[dict] = json.loads(
+            self._bucket.get_blob("hisanta_prompts.json").download_as_bytes()
+        )
+        """List of system prompts for each agent, with the time at which each became effective.
+        These all look like {"agent_id": AGENT_ID, "prompts": [{"start": ISO_TIMESTAMP, "prompt": PROMPT}]}
+        where the prompts are in chronological order."""
+
+    def __iter__(self) -> Generator[VoiceSample, Any, None]:
+        worker_info = data.get_worker_info()
+        start = worker_info.id if worker_info else 0
+        increment = worker_info.num_workers if worker_info else 1
+        for i in range(start, len(self._conversations), increment):
+            yield from self._from_conversation(i)
```

Comment:
We'll probably have to experiment with how to form our samples here.
There are multiple issues to consider:
1. How to do shuffle
2. The length of each sample should be regulated: `max_audio_duration_secs` was an attempt at this, but generally the bottleneck is GPU memory